### ISSUE TITLE
dnf5 remove: also support `package-spec`?

### ISSUE THREAD

**Issue by sanjayankur31** [rpm-software-management/dnf5#1958]
I was looking at removing some packages based on their provides and see that `dnf remove` does not support this.

A very important use case here is TeXLive packages, where it's easier to install the required packages based on their provides. For example, one can run `sudo dnf install tex(cleveref.sty)` and it'll install the package that provides this style file. Now, once one is done with a TeXlive project, it's useful to remove these packages because they tend to take up quite a lot of space. However, but one cannot currently use `dnf remove tex(cleveref.sty)` because the remove command only takes `package-file-spec` and not `package-spec`.

I'm sure there's a reason why `dnf remove` and `dnf install` don't accept the same package specification, but it's not intuitive that one can specify provides for installation but not for removal of the packages. Would it please be possible to make `dnf remove` also accept `package-spec`?

**Comment by ppisar** [comment]
Duplicate of #1937.

**Comment by sanjayankur31** [comment]
F41, so definitely dnf5.
